### PR TITLE
Master better search in settings qmo

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
UPDATE ME

https://www.odoo.com/web#id=1893252&action=333&active_id=965&model=project.task&view_type=form&menu_id=4720

### CODE REVIEW

**Comment by xmo-odoo** [comment]
> UPDATE ME

4c1

**Comment by qmo-odoo** [comment]
Purpose :  
Allows users to not only search on settings name but also on their description.
Improves the results order : Top result should be about the selected module.

Current behavior before PR:
Users can only search on settings name and the results order is fixed no matter the module you're in.

Desired behavior after PR:
Users can search on descriptions  and top results are about the module they selected in the left panel

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ var FormController = require('web.FormController');
 var FormRenderer = require('web.FormRenderer');
 var view_registry = require('web.view_registry');
 
+
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: useless noise

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 2]
```diff
@@ -21,17 +22,17 @@ var BaseSettingRenderer = FormRenderer.extend({
         this._super.apply(this, arguments);
         this.activeView = false;
         this.activeTab = false;
+        this.initial_order;
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: this.initialTabElements = undefined;

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 3]
```diff
@@ -21,17 +22,17 @@ var BaseSettingRenderer = FormRenderer.extend({
...
     },
-
+
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: noise, remove this

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 4]
```diff
@@ -21,17 +22,17 @@ var BaseSettingRenderer = FormRenderer.extend({
...
-
+    
     start: function () {
         this._super.apply(this, arguments);
         if (config.device.isMobile) {
             core.bus.on("DOM_updated", this, function () {
                 this._moveToTab(this.currentIndex || this._currentAppIndex());
             });
-        }
+        }
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: noise, remove this

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 5]
```diff
@@ -236,7 +248,7 @@ var BaseSettingRenderer = FormRenderer.extend({
 
     _onKeyUpSearch: function (event) {
         this.searchText = this.searchInput.val();
-        this.activeTab.removeClass('selected');
+        // this.activeTab.removeClass('selected');
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: remove this

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 6]
```diff
@@ -199,6 +200,17 @@ var BaseSettingRenderer = FormRenderer.extend({
      * @param {int} index
      */
     _moveToTab: function (index) {
+        //Used to reorder the settings after a search and a tab switch
+        //Thought it would be better for the UX if the order stayed the same as the one on the left panel
+        if(!this.initial_order && this.$el.find('.settings')[0]){
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: you use 3 times the this.$el.find('.settings')[0] expression.  Also a lookup in the DOM is kind of expensive. You can simply do this:
```
var settingsNode = this.$('.settings')[0];
if (!this.initialTabElements && settingsNode[0]) {...
```

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 7]
```diff
@@ -199,6 +200,17 @@ var BaseSettingRenderer = FormRenderer.extend({
...
+            //Converts an HtmlCollection into an array
+            this.initial_order = Array.prototype.slice.call(this.$el.find('.settings')[0].children);
+        }
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: else on same line as }

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 8]
```diff
@@ -199,6 +200,17 @@ var BaseSettingRenderer = FormRenderer.extend({
...
+        else{
+            var current_order = this.$el.find('.settings');
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]:
```
this.$('.settings')
    .empty()
    .append(this.initialTabElements)
```

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 9]
```diff
@@ -319,7 +337,7 @@ var BaseSettingRenderer = FormRenderer.extend({
                 if (self.inVisibleCount !== resultSetting.length) {
                     module.settingView.find('.settingSearchHeader').removeClass('o_hidden');
                     module.settingView.removeClass('o_hidden');
-                }
+                }
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: remove this

File: odoo/addons/base/static/src/js/res_config_settings.js [hunk 10]
```diff
@@ -299,13 +311,19 @@ var BaseSettingRenderer = FormRenderer.extend({
     _searchSetting: function () {
         var self = this;
         this.count = 0;
+        var settings = $(this.$el.find('.settings')[0].children)
```
Comment (ged-odoo) [inline odoo/addons/base/static/src/js/res_config_settings.js]: ; at the end of this (and the next) lines

**Comment by ged-odoo** [comment]
ok (but well, not really happy with the general state/quality of the settings form view. I think that the code is doing stuff the 'wrong' way in general, and that it should have been done differently, and that it should be optimized (thinking about the name_gets))

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**